### PR TITLE
[SMALLFIX] Improve runUfsTests console message and tests

### PR DESCRIPTION
**Pull request by LuQQiu**
Previously, the `./bin/alluxio runUfsTests` will log all the messages on the console. When running against object stores, many retry messages will show on the console to make users confusing. Moving the logs to user.log. Fix one UFS test.

### CODE REVIEW

**Comment by LuQQiu** [comment]
@madanadit PTAL, thanks

File: core/common/src/main/java/alluxio/util/CommonUtils.java [hunk 1]
```diff
@@ -70,6 +71,43 @@
...
+   * @param stream the print stream to write to
+   * @return the thread
+   */
+  public static Thread createProgressThread(PrintStream stream) {
+    return createProgressThread(2 * Constants.SECOND_MS, stream);
+  }
+
+  /**
+   * Creates a thread which will write "." to the given print stream at the given interval. The
+   * created thread is not started by this method. The created thread will be daemonic and will
```
Comment (madanadit) [inline core/common/src/main/java/alluxio/util/CommonUtils.java]: nit: daemonic=demonic="A persistently tormenting person"

File: core/common/src/main/java/alluxio/util/CommonUtils.java [hunk 2]
```diff
@@ -70,6 +71,43 @@
...
+   * halt when interrupted.
+   *
+   * @param intervalMs the time interval in milliseconds between writes
+   * @param stream the print stream to write to
+   * @return the thread
+   */
+  public static Thread createProgressThread(final long intervalMs, final PrintStream stream) {
+    Thread thread = new Thread(new Runnable() {
```
Comment (madanadit) [inline core/common/src/main/java/alluxio/util/CommonUtils.java]: nit: you could use lambda as
```
Thread t = new Thread(() -> {
    // your code here ...
});
```

File: examples/src/main/java/alluxio/cli/UnderFileSystemContractTest.java [hunk 3]
```diff
@@ -125,16 +124,20 @@ private void loadAndRunTests(Object operations, String testDir) throws Exception
       for (Method test : tests) {
         String testName = test.getName();
         if (testName.endsWith("Test")) {
-          LOG.info("Running test: " + testName);
+          System.out.printf("Running test: %s...", testName);
+          Thread thread = CommonUtils.createProgressThread(2 * Constants.SECOND_MS, System.out);
```
Comment (madanadit) [inline examples/src/main/java/alluxio/cli/UnderFileSystemContractTest.java]: nit: why not use the wrapper?

File: examples/src/main/java/alluxio/examples/UnderFileSystemCommonOperations.java [hunk 4]
```diff
@@ -237,15 +237,23 @@ public void createOpenExistingLargeFileTest() throws IOException {
...
-    if (buf.length != bytesRead) {
-      throw new IOException(FILE_CONTENT_INCORRECT);
-    }
-    for (int i = 0; i < bytesRead; ++i) {
-      if (TEST_BYTES[i % TEST_BYTES.length] != buf[i]) {
-        throw new IOException(FILE_CONTENT_INCORRECT);
+    int offset = 0;
+    while (offset < buf.length) {
+      int bytesRead = inputStream.read(buf, offset, buf.length - offset);
+      if (bytesRead != -1) {
```
Comment (madanadit) [inline examples/src/main/java/alluxio/examples/UnderFileSystemCommonOperations.java]: nit: break first and remove indent for the if block

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java [hunk 5]
```diff
@@ -99,7 +101,7 @@ private void load(AlluxioURI filePath, int replication)
         load(newPath, replication);
       }
     } else {
-      Thread thread = JobGrpcClientUtils.createProgressThread(System.out);
+      Thread thread = CommonUtils.createProgressThread(2 * Constants.SECOND_MS, System.out);
```
Comment (madanadit) [inline shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: nit: use wrapper

File: shell/src/main/java/alluxio/cli/fsadmin/command/CheckpointCommand.java [hunk 6]
```diff
@@ -41,7 +41,7 @@ public String getCommandName() {
 
   @Override
   public int run(CommandLine cl) throws IOException {
-    Thread thread = JobGrpcClientUtils.createProgressThread(2 * Constants.SECOND_MS, System.out);
+    Thread thread = CommonUtils.createProgressThread(2 * Constants.SECOND_MS, System.out);
```
Comment (madanadit) [inline shell/src/main/java/alluxio/cli/fsadmin/command/CheckpointCommand.java]: same here

File: core/common/src/main/java/alluxio/util/CommonUtils.java (hunk 1 above)
Comment (LuQQiu) [reply core/common/src/main/java/alluxio/util/CommonUtils.java]: Change to a daemon thread, Thanks for explanation

File: core/common/src/main/java/alluxio/util/CommonUtils.java (hunk 2 above)
Comment (LuQQiu) [reply core/common/src/main/java/alluxio/util/CommonUtils.java]: Done

File: examples/src/main/java/alluxio/cli/UnderFileSystemContractTest.java (hunk 3 above)
Comment (LuQQiu) [reply examples/src/main/java/alluxio/cli/UnderFileSystemContractTest.java]: Changed

File: examples/src/main/java/alluxio/examples/UnderFileSystemCommonOperations.java (hunk 4 above)
Comment (LuQQiu) [reply examples/src/main/java/alluxio/examples/UnderFileSystemCommonOperations.java]: Changed

File: shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java (hunk 5 above)
Comment (LuQQiu) [reply shell/src/main/java/alluxio/cli/fs/command/DistributedLoadCommand.java]: Changed

File: shell/src/main/java/alluxio/cli/fsadmin/command/CheckpointCommand.java (hunk 6 above)
Comment (LuQQiu) [reply shell/src/main/java/alluxio/cli/fsadmin/command/CheckpointCommand.java]: Changed

**Comment by LuQQiu** [comment]
@calvinjia PTAL, addressed all the comments @madanadit left. Thanks

File: examples/src/main/java/alluxio/cli/UnderFileSystemContractTest.java [hunk 7]
```diff
@@ -125,16 +123,20 @@ private void loadAndRunTests(Object operations, String testDir) throws Exception
...
+          thread.start();
           try {
             test.invoke(operations);
           } catch (InvocationTargetException e) {
             if (mUfs.getUnderFSType().equals(S3_IDENTIFIER)) {
               logRelatedS3Operations(test);
             }
             throw new IOException(e.getTargetException());
+          } finally {
+            thread.interrupt();
```
Comment (calvinjia) [inline examples/src/main/java/alluxio/cli/UnderFileSystemContractTest.java]: do we always want to interrupt the thread?

Comment (LuQQiu) [reply examples/src/main/java/alluxio/cli/UnderFileSystemContractTest.java]: Actually we do need to thread to print dot. 
The thread will be created and interrupted in a short period if we run tests against Hdfs or local UFS.

Comment (LuQQiu) [reply examples/src/main/java/alluxio/cli/UnderFileSystemContractTest.java]: Removed the print dot thread.

**Comment by LuQQiu** [comment]
@calvinjia PTAL, thanks

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**